clients/unshaded/src/main/java/tachyon/client/file/FileSystem.java

```diff
@@ -0,0 +1,375 @@
+      throws FileDoesNotExistException, IOException, TachyonException;
+
+  /**
+   * Gets the {@link URIStatus} object that represents the metadata of a Tachyon path.
+   *
+   * @param path the path to obtain information about
+   * @param options options to associate with this operation
+   * @return the {@link URIStatus} of the file
+   * @throws IOException if a non-Tachyon exception occurs
+   * @throws FileDoesNotExistException if the path does not exist
+   */
+  URIStatus getStatus(TachyonURI path, GetStatusOptions options)
+      throws FileDoesNotExistException, IOException, TachyonException;
+
+  /**
+   * Convenience method for {@link #listStatus(TachyonURI, ListStatusOptions)} with default options.
+   *
+   * @param path the path to list information about
+   * @return a list of {@link URIStatus}s containing information about the files and directories
+   *         which are children of the given path
+   * @throws IOException if a non-Tachyon exception occurs
+   * @throws FileDoesNotExistException if the given path does not exist
+   * @throws TachyonException if an unexpected Tachyon exception is thrown
+   */
+  List<URIStatus> listStatus(TachyonURI path)
+      throws FileDoesNotExistException, IOException, TachyonException;
+
+  /**
+   * If the path is a directory, returns the {@link URIStatus} of all the direct entries in it.
+   * Otherwise returns the {@link URIStatus} for the file.
+   *
+   * @param path the path to list information about
+   * @param options options to associate with this operation
+   * @return a list of {@link URIStatus}s containing information about the files and directories
+   *         which are children of the given path
+   * @throws IOException if a non-Tachyon exception occurs
+   * @throws FileDoesNotExistException if the given path does not exist
+   * @throws TachyonException if an unexpected Tachyon exception is thrown
+   */
+  List<URIStatus> listStatus(TachyonURI path, ListStatusOptions options)
+      throws FileDoesNotExistException, IOException, TachyonException;
+
+  /**
+   * Convenience method for {@link #loadMetadata(TachyonURI, LoadMetadataOptions)} with default
+   * options.
+   *
+   * @param path the path for which to load metadata from UFS
+   * @return the {@link TachyonURI} referencing the new path available in Tachyon
+   * @throws IOException if a non-Tachyon exception occurs
+   * @throws FileDoesNotExistException if the given path does not exist
+   * @throws TachyonException if an unexpected Tachyon exception is thrown
+   */
+  TachyonURI loadMetadata(TachyonURI path)
+      throws FileDoesNotExistException, IOException, TachyonException;
+
+  /**
+   * Loads metadata about a path in the UFS to Tachyon. No data will be transferred.
+   *
+   * @param path the path for which to load metadata from UFS
+   * @param options options to associate with this operation
+   * @return the {@link TachyonURI} referencing the new path available in Tachyon
```

Comment:
@jsimsa I don't think it would be different. What kind of return value do you prefer? I think `void` may be the other option, boolean is a bit messy since we need to define when `false` is returned as opposed to an exception.